source/gmsh/utilities.cc

```diff
@@ -25,6 +25,17 @@
 
 #ifdef DEAL_II_WITH_GMSH
 
+
+#  ifdef DEAL_II_GMSH_WITH_API
+#    include <deal.II/base/config.h>
```

Comment:
This include isn't necessary. You'll get it through any of the other header files.